src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,138 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import org.testng.annotations.Test;
+
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    private Scanner scanner;
+    List<String> tests = new ArrayList<String>();
+    List<String> directoriesTested = new ArrayList<String>();
+
+    @Test
+    public void checksTestsInTestNg() {
+        
+        // Verify that testng.xml exists
+        File f = new File("./src/test/testng.xml");   
+        assertTrue(f.exists() && !f.isDirectory());
+        
+        updateDirectoriesTested();
+        getTestFiles("./src/test/java/teammates/test");
```

Comment:
I think you can start from the `./src/test/java/teammates/test/cases` directory. I've made it such that all test cases are in the subfolders of that directory.